projects/plugins/wpcomsh/wpcom-migration-helpers/site-migration-helpers.php

```diff
@@ -115,3 +115,38 @@ function ( $should_notify ) {
 	);
 }
 add_action( 'plugins_loaded', 'aiowp_migration_logging_helper', 10 );
+
+/**
+ * Helper function that registers a filter that listens for the AIOWP migration completed event.
+ * Once detected, it will trigger a call to an endpoint on WPCOM to run the corresponding cleanup jobs.
+ */
+function aiowp_migration_status_helper() {
+	if ( ! class_exists( 'Ai1wm_Main_Controller' ) ) {
+		return;
+	}
+
+	$wpcom_blog_id = _wpcom_get_current_blog_id();
+	add_filter(
+		'ai1wm_import',
```

Comment:
Do you know where this is triggered?

Reply:
I'm wondering if it's semantically correct to use a filter here, since filters are generally meant to modify and return data. In this case, since we're invoking an endpoint and not transforming data, this feels more appropriate as an action.

Reply:
This is triggered from AIOWP plugin. Whether it's semantically correct or not is not that important, since there's no other know way to detect when the migration is finished. We got this info from ServMask themselves (you can check it out in the description of the ticket DOTCOM-1799)